app/parse.py

```diff
@@ -8,8 +15,53 @@
     tags: list[str]
 
 
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+
+
+def parse_single_quote(quote) -> Quote:
+    return Quote(
+        text=quote.select_one(".text").text,
+        author=quote.select_one(".author").text,
+        tags=[tag.text for tag in quote.select(".tag")],
+    )
+
+
+def get_single_page_quotes(page_soup) -> [Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote) for quote in quotes]
+
+
+def get_all_quotes():
+    page_num = 1
+    all_quotes = []
+
+    while True:
+        page = requests.get(f"{BASE_URL}/page/{page_num}").content
+        page_soup = BeautifulSoup(page, "html.parser")
+
+        quotes = get_single_page_quotes(page_soup)
+        all_quotes.extend(quotes)
+
+        if not any("next" in page_link["class"] for page_link in
+                   page_soup.select(".pager > li")):
```

Comment:
The logic for checking if there is a 'next' page might not work as intended. The condition `if not any('next' in page_link['class'] for page_link in page_soup.select('.pager > li')):` assumes that the 'next' button will always have a class containing 'next'. Ensure this is the case by inspecting the HTML structure of the website.